### PR TITLE
Feat/create reservation

### LINKED ISSUE #17: Create reservation page
A page containing a form where the user can fill out reservation details.

### PR DESCRIPTION
**Pull request by MindChirp**
Page for creating a reservation.

The following features have been implemented:
- Form with field validation using react-hook-form (through shadcn)
- Confirm modal which appears before submitting the form
- User type, which will be used when we want to store user data that has been fetched from Lepton
- A "Sender inn forespørsel som" placeholder element. The idea is that it will display the username and profile picture, assuring the user that they are sending a reservation request through their own user.
- Some query param parsing, which populates the form with data. `from` and `to` date values have not been implemented yet. `itemUUID` will however be parsed, and as a result the corresponding item will be automatically selected in the form.
- A new, custom datetime picker
- Custom error page which I suggest that we start using in other parts of the website as well. It is currently only shown if errors occur within in the "/booking/" URL-path. 
- **Created a `max-w-page` tw-class. This will normally set the max-width equals `7xl`, but if the screen becomes sufficiently small, it will default to 90% of the parent wrapper width. This creates some padding on each side of the page content, which makes everything more pleasant to view on small devices. View the 4th attached image to know what this looks like.**

_**PS: When testing the page, be sure to have added some bookable items in the django admin panel. If there are no bookable items, you will be taken to the error page.**_

image
image
image
image

### CODE REVIEW

**Comment by eriskjel** [comment]
litt smalere content boks eller?
ser ellers veldig bra ut
considering...

**Comment by MindChirp** [comment]
> litt smalere content boks eller?
> ser ellers veldig bra ut
> considering...

Hvilken boks?

**Comment by eriskjel** [comment]
> > litt smalere content boks eller?
> > ser ellers veldig bra ut
> > considering...
> 
> Hvilken boks?

hovedboksen lism, under h1

**Comment by MindChirp** [comment]
> > > litt smalere content boks eller?
> > > ser ellers veldig bra ut
> > > considering...
> > 
> > 
> > Hvilken boks?
> 
> hovedboksen lism, under h1

min broder i jesus kristus, dette fikser vi etter at mvp-en er ferdig

File: src/app/booking/components/ReservationFormFields.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+type EventFormFieldsType = {
+    initialItem: string;
+    items: DetailedItem[];
+    onSubmit: (values: EventFormValueTypes) => void;
+}
+
+/**
+ * Component that contains all of the fields that belong to the reservation form.
+ */
+const EventFormFields = ({initialItem, items, onSubmit}: EventFormFieldsType) => {
```
Comment (ConradOsvik) [inline src/app/booking/components/ReservationFormFields.tsx]: burde ikke denne hete ReservationFormFields?

**Review by ConradOsvik — COMMENTED** [review]
ser bra ut

**Pull request merged**